### PR TITLE
0.7 compat WIP

### PR DESCRIPTION
**Pull request by Keno**

### CODE REVIEW

File: src/clanginstances.jl [hunk 1]
```diff
@@ -16,7 +16,11 @@ destructs = Dict{ClangCompiler,Function}()
 
 function get_destruct_for_instance(C)
     if !haskey(destructs, C)
-        idx = findfirst(active_instances, C)
+        if VERSION <= v"0.7-"
+            idx = findfirst(active_instances, C)
+        else
+            idx = findfirst(equalto(C), active_instances)
```
Comment (ararslan) [inline src/clanginstances.jl]: This method is available in Compat for previous versions, so you can bump the required version of Compat and remove this version check altogether.

Comment (Keno) [reply src/clanginstances.jl]: I don't really like Compat. With femtocleaner removing the check automatically once I bump to 0.7, I find this way much clearer.

File: test/cxxstr.jl [hunk 2]
```diff
@@ -1,5 +1,10 @@
 using Cxx
-using Base.Test
+if VERSION <= v"0.7-"
+    using Base.Test
+else
+    using Test
```
Comment (ararslan) [inline test/cxxstr.jl]: You can use `Compat.Test`

File: src/cxxstr.jl [hunk 3]
```diff
@@ -393,7 +393,7 @@ function find_expr(sourcebuf,str,pos = 1)
     end
     write(sourcebuf,str[pos:(idx-1)])
     # Parse the first expression after `$`
-    expr,pos = parse(str, idx + 1; greedy=false)
+    expr,pos = Meta.parse(str, idx + 1; greedy=false)
```
Comment (ararslan) [inline src/cxxstr.jl]: Isn't this only a thing in 0.7?

Comment (Keno) [reply src/cxxstr.jl]: Yes, will fix

File: test/std.jl [hunk 4]
```diff
@@ -226,8 +237,9 @@ end
         @test convert(Vector{Bool}, cxx_bool_v) == jl_bool_v
 
         let x = [true, false, true]
-            y = convert(cxxt"std::vector<$Int>", x)
-            @test collect(y) == Int[1, 0, 1]
+            let y = convert(cxxt"std::vector<$Int>", x)
```
Comment (ararslan) [inline test/std.jl]: Why is the `let` necessary here? It's already inside of a `let` block, so `y` is still constrained to that scope, right?

Comment (Keno) [reply test/std.jl]: Avoids a deprecation warning. There's a global scope `y` somewhere.

File: src/clanginstances.jl (hunk 1 above)
Comment (ararslan) [reply src/clanginstances.jl]: You already require Compat though: https://github.com/Keno/Cxx.jl/blob/master/REQUIRE#L2. I mean it's your code so what I think here doesn't really matter, but what about unconditionally writing `findfirst(equalto(C), active_instances)` do you find unclear?

Comment (Keno) [reply src/clanginstances.jl]: I'd like to get rid of the Compat dependency.

Comment (ararslan) [reply src/clanginstances.jl]: Okay. Sorry for the noise here, just excited about 0.7 compatibility.

**Comment by Keno** [comment]
Linux CI seems to be borked, but that's for another PR to figure out.

**Pull request merged**